core/server/master/src/main/java/alluxio/master/file/DefaultFileSystemMaster.java

```diff
@@ -1486,9 +1486,27 @@ public void delete(AlluxioURI path, DeleteOptions options) throws IOException,
         LockedInodePath inodePath =
             mInodeTree.lockInodePath(lockingScheme.getPath(), lockingScheme.getMode());
         FileSystemMasterAuditContext auditContext =
-            createAuditContext("delete", path, null, inodePath.getInodeOrNull())) {
+            createAuditContext("delete", path, null, inodePath.getInodeOrNull());
+        LockedInodePathList children = !options.isRecursive() ? null
```

Comment:
easier to read if we flip the order of the ternary so that we don't need to negate the condition